pandas/tests/io/test_pickle.py

```diff
@@ -190,29 +196,27 @@ def python_unpickler(path):
     ],
 )
 @pytest.mark.parametrize("writer", [pd.to_pickle, python_pickler])
-def test_round_trip_current(current_pickle_data, pickle_writer, writer):
-    data = current_pickle_data
-    for typ, dv in data.items():
-        for dt, expected in dv.items():
-            with tm.ensure_clean() as path:
-                # test writing with each pickler
-                pickle_writer(expected, path)
-
-                # test reading with each unpickler
-                result = pd.read_pickle(path)
-                compare_element(result, expected, typ)
-
-                result = python_unpickler(path)
-                compare_element(result, expected, typ)
-
-                # and the same for file objects (GH 35679)
-                with open(path, mode="wb") as handle:
-                    writer(expected, path)
-                    handle.seek(0)  # shouldn't close file handle
-                with open(path, mode="rb") as handle:
-                    result = pd.read_pickle(handle)
-                    handle.seek(0)  # shouldn't close file handle
-                compare_element(result, expected, typ)
+@pytest.mark.parametrize("typ, expected", flatten(create_pickle_data()))
```

Comment:
how big is the memory footprint of create_pickle_data?  i think doing this outside the test means it sticks around during the whole collection/runtime

Reply:
Doing a recursive `getsizeof` of the dict yields 426965 bytes so it doesn't seem too much. Additionally IIUC before this was being yielded from the `current_pickle_data` fixture which caches & keeps the data around after first use